Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm

```diff
@@ -285,44 +219,15 @@ void requestLinkDecorationFilteringData(LinkFilteringRulesCallback&& callback)
     }];
 }
 
-StorageAccessPromptQuirkController& StorageAccessPromptQuirkController::shared()
-{
-    static MainThreadNeverDestroyed<StorageAccessPromptQuirkController> sharedInstance;
-    return sharedInstance.get();
-}
-
-Ref<StorageAccessPromptQuirkObserver> StorageAccessPromptQuirkController::observeUpdates(Function<void()>&& completionHandler)
+WPResourceType StorageAccessPromptQuirkController::resourceType() const
 {
-    ASSERT(RunLoop::isMain());
-    if (!m_notificationListener) {
-        m_notificationListener = adoptNS([WKWebPrivacyNotificationListener new]);
-        [m_notificationListener listenForStorageAccessPromptQuirkChanges:^{
-            updateQuirks([this] {
-                m_observers.forEach([](auto& observer) {
-                    observer.invokeCallback();
-                });
-            });
-        }];
-    }
-    Ref observer = StorageAccessPromptQuirkObserver::create(WTFMove(completionHandler));
-    m_observers.add(observer.get());
-    return observer;
+    return static_cast<WPResourceType>(WPResourceTypeStorageAccessPromptQuirksData);
```

Comment:
Is there a need to cast here? Looking at `LinkDecorationFilteringController::resourceType()` where we don't.

Reply:
Unfortunately, I think we need a cast to keep the non-internal builds passing :/ (see the failing builds from earlier)